### PR TITLE
Fixed a couple of bugs on the systemd service for keepalive

### PR DESCRIPTION
**Pull request by mssola**
## Description

There were two issues reported against the systemd unit for the keepalive service. Namely:

1. A customer told us that the keepalive service was ignoring a configured proxy. Proxies use the `/etc/sysconfig/proxy` file, and so the provided `.service` file from our package must include it as an optional environment file. The first commit fixes this.
2. The service file required the `multi-user` target, which would tell systemd to run this service whenever a new shell user is set up. This was not wanted either, so I have removed it on the second commit.

With these two updates, I have updated the version of `SUSEConnect` to `0.4` according to the new versioning scheme that we have discussed internally.

## How to test

The easiest way to reproduce [bsc#1200994](https://bugzilla.suse.com/show_bug.cgi?id=1200994) is a bit tricky. In my case I started a VM with SLE 15 SP4 (it cannot be a docker container because we need systemd in order to test this, but it can also be a random host you have). Then:

1. Install the latest SUSEConnect from OBS and apply my patch (just copy-paste the `.service` file).
2. Run `systemd daemon-reload`.
3. Register the machine as usual.
4. Run `systemctl start suseconnect-keepalive.service` and check that it works as usual.
5. Apply the configuration you can find in the related bug for `/etc/sysconfig/proxy`.
6. Run `systemctl start suseconnect-keepalive.service` again and check that it doesn't work (because you do not have access to the proxy as given on that configuration).
7. Run `systemctl status suseconnect-keepalive.service` just to check that it failed because it tried to reach that unreachable proxy. This is proof enough that it picks the values from that file.

## Related bugs

Fixes [bsc#1200994](https://bugzilla.suse.com/show_bug.cgi?id=1200994).

### CODE REVIEW

**Comment by mssola** [comment]
[[URAS] suseconnect-keepalive.service doesn't use proxy env vars](https://trello.com/c/1ePLcow0/2476-uras-suseconnect-keepaliveservice-doesnt-use-proxy-env-vars)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/528484/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged dd59bdf27216a78977fa34eca0c5fc15473cbb11 into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/528704/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 9068d44e99aa18f7e21f6973b42e4896da748fc0 into 09e481808676027e88874b57f4481dc53f81d3e9)

File: lib/suse/connect/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module SUSE
   module Connect
-    VERSION = '0.3.35'
+    VERSION = '0.4'
```
Comment (lagartoflojo) [inline lib/suse/connect/version.rb:3]: According to the [semver docs](https://semver.org/#spec-item-2), I believe we'd need to make the version `0.4.0`

Comment (mssola) [reply lib/suse/connect/version.rb:3]: That's part of the discussion we had with @ngetahun, in which for our products it doesn't make quite much sense to follow semver exactly. I believe Nate is to update the docs on our side, but the patch-level release number is no longer needed.

Comment (lagartoflojo) [reply lib/suse/connect/version.rb:3]: Okay, thanks for the info! Seems I missed it

**Comment by digitaltom** [comment]
Please see the discussion here: https://github.com/SUSE/connect-ng/pull/140/files 

It seems due to 
```
[Install]
WantedBy=multi-user.target
```
we run the keepalive on each boot. Let's discuss this with the team, I think this is an unwanted feature.

**Comment by mssola** [comment]
> Please see the discussion here: https://github.com/SUSE/connect-ng/pull/140/files 
> we run the keepalive on each boot. Let's discuss this with the team, I think this is an unwanted feature.

Done!

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/530129/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 35eda7ed991e1d7c17411c6da4949d6a0d18731c into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/530710/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 8f80b9e73a510acbcebabd5eb4c066d587669a7d into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531843/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 4c154f6bb6ac1d3943826f8514891d4703465ecd into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531844/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged e722db3c9c99dc12eb8f3e1e9bb15f6bf42f060a into 09e481808676027e88874b57f4481dc53f81d3e9)

File: lib/suse/connect/connection.rb [hunk 2]
```diff
@@ -57,6 +58,27 @@ def initialize(endpoint, language: nil, insecure: false, debug: false, verify_ca
 
       private
 
+      # Returns true if a specific configuration on the system allows or not to
+      # use a configured proxy.
+      #
+      # This is usually already handled by the NET::HTTP library, but it only
+      # checks for some very specific environment variables (see
+      # URI::HTTP#find_proxy?). In some environments there might be the
```
Comment (digitaltom) [inline lib/suse/connect/connection.rb:66]: let's mention here that this comes from /etc/sysconfig/proxy?

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531846/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged d643f04e4e765d4fc30e666beee52ac0ef0a488d into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531848/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 2228fe03d38a51c71d9baa7f5b023915843776bb into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531855/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 80d382e2858df9c78d02980bdf201be2fe570b41 into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531856/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 189d0ab249a185b29f4e164bc978eaceab1a5812 into 09e481808676027e88874b57f4481dc53f81d3e9)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/531857/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged b2758adacf2092cbbef8b8286c201a7d1ba5e7e1 into fae8b391bb895b2a745a19f9d7f61caa00b063a6)

**Review by digitaltom — APPROVED** [review]

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/531860/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6944b95c0c4decb1cf34079d994c36ffd24df91d into fae8b391bb895b2a745a19f9d7f61caa00b063a6)

**Pull request merged**